templates/etcd-ha/3/README.md

```diff
@@ -1,41 +1,38 @@
+### Upgrades
 
-Scaling up an existing etcd cluster is fully automated using the [Etcd Members API](https://coreos.com/etcd/docs/2.3.0/members_api.html).
+Starting with `2.3.6-rancher4`, upgrades are fully supported and require no user intervention beyond navigating the UI and selecting the desired version. In a standalone deployment, service downtime is inevitable. In a 3-node deployment, service downtime may be experienced as quorum is lost for a brief period of time. This is a limitation of the template and will be addressed in future versions. 5+ node upgrades should see no downtime.
 
-Scaling down is unsupported..
+### Resiliency
 
+Etcd can survive `floor(N/2)` recoverable or unrecoverable failures while maintaining 100% uptime. For recoverable host failures such as a power cycle, the service self-heals. For unrecoverable host failures, the service self-heals when sufficient resources are allocated to the environment. Therefore, it is not a bad idea to allocate an extra host beyond the specified cluster size. For instance, a 3-node deployment in a 4-host environment may survive 2 host failures, given enough time in-between to self-heal the cluster. The amount of time which must pass between host failures is indeterminate and depends on how much data must be replicated to the new node.
 
-### Usage
+Etcd can survive `N` recoverable failures, but system downtime will be experienced once a majority of nodes (quorum) is lost.
 
-Select etcd the catalog page.
+Etcd can survive `N-1` unrecoverable failures, but will enter a disaster state where some user intervention is required to recover. Recovering from this situation is mostly-automated, but will not be fully automated. This is a very, very bad situation to be in and should never occur if appropriate steps are taken such as spreading hosts across availability zones.
 
-Fill in the number of nodes desired. This should be an **ODD** number. Recommended configurations are 3, 5, or 7 node deployments. More nodes increases read availability while decreasing read latency. Less nodes decreases write latency, but sacrifices read latency and availability.
+### Disaster Recovery (DR)
 
-Click deploy.
+If a majority of nodes are unrecoverably lost, you must re-build the cluster from one of the surviving nodes. The process involves selecting a survivor, transforming it into a standalone node, and adding new hosts so the service may scale back up to the desired size.
 
-Once the stack is deployed and assuming your application is deployed within it, you can access the etcd cluster in your application like so:
+In more detail, follow these steps:
 
-```
-etcdctl --endpoints http://etcd:2379 member list
-```
+1. Determine if your lost hosts/containers are truly unrecoverable. If a host comes back online, wait at least 5 minutes while Network Agent repairs itself and the Etcd container should return to a running state. If the hosts are truly unrecoverable, remove them from the environment.
```

Comment:
is there an easy/obvious way to determine which containers are useless vs which are candidates for recovery? it might be good to make this more clear.